src/hotspot/share/runtime/thread.cpp

```diff
@@ -1720,7 +1718,8 @@ void JavaThread::send_thread_stop(oop java_throwable)  {
       }
 
       // Set async. pending exception in thread.
-      set_pending_async_exception(java_throwable);
+      _pending_async_exception = java_throwable;
+      set_suspend_flag(_has_async_exception);
```

Comment:
Not clear why we didn't just simplify set_pending_async_exception.

Reply:
Right, maybe is better to do that. I restored set_pending_async_exception() and simplified it instead.